my-awesome-rabbitmq-lib/src/events.rs

```diff
@@ -164,31 +194,31 @@ impl PayloadEvent for LegendMissionsOngoingMissionEventPayload {
     }
 }
 
-#[derive(Debug, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct CoinsNotifyClientPayload {
-    pub room: String,
-    pub message: HashMap<String, serde_json::Value>,
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "PascalCase")]
+pub enum RewardType {
+    Legends,
+    CodeExchange,
 }
 
-impl PayloadEvent for CoinsNotifyClientPayload {
-    fn event_type(&self) -> MicroserviceEvent {
-        MicroserviceEvent::CoinsNotifyClient
-    }
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RankingWinner {
+    #[serde(rename = "userId")]
+    pub user_id: String,
+    pub reward: i32,
 }
 
 #[derive(Debug, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct CoinsSendEmailPayload {
-    pub user_id: String,
-    pub email_type: String,
-    pub email: String,
-    pub coins: i32,
+pub struct LegendRankingsRankingsFinishedEventPayload {
+    pub title: String,
+    #[serde(rename = "rewardType")]
+    pub reward_type: RewardType,
+    pub winners: Vec<RankingWinner>,
 }
```

Comment:
Este no es el payload, el payload que declaraste en la librería de TS es `  completedRankings: CompletedRanking[];`, cambia la librería de TS

Reply:
Rigth, here is the correction:
```.ts
// ts

// interface
export interface CompletedRanking {
    title: string;
    rewardType: RankingsRewardsType;
    winners: RankingWinners[];
}

// event
'legend_rankings.rankings_finished': {
      completedRankings: CompletedRanking[];
};
```

```rust
// .rs

// struct
#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CompletedRanking {
    pub title: String,
    pub reward_type: RankingsRewardsType,
    pub winners: Vec<RankingWinners>,
}

// event
#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LegendRankingsRankingsFinishedEventPayload {
    pub completed_rankings: Vec<CompletedRanking>,
}